openassessment/xblock/static/js/src/lms/oa_datefactory.js

```diff
@@ -0,0 +1,90 @@
+
+/**
+ *
+ * A helper function to utilize DateUtils.
+ **/
+
+OpenAssessment.DateTimeFactory = function(element) {
+    this.element = element;
+};
+
+OpenAssessment.DateTimeFactory.prototype = {
+
+    report: function() {
+        var opFactory = this;
+        $('.ora-datetime', this.element).each(function() {
+            opFactory.generate($(this));
+        });
+    },
+
+    determineContext: function(el) {
+        var context;
+        context = {
+            datetime: el.data('datetime'),
+            timezone: el.data('timezone'),
+            language: el.data('language'),
+            format: ''
+        };
+        return context;
+    },
+
+    determineDateToken: function(el) {
+        var opFactory = this;
+        var dateToken = 'date';
+        if (opFactory.isValid(el.data('datetoken'))) {
+            dateToken = el.data('datetoken');
+        }
+        return dateToken;
+    },
+
+    generate: function(el) {
+        var opFactory = this;
+        (function(require) {
+            require([
```

Comment:
It is unfortunate to have the only usage of ``require`` be hidden inside a function like this, but in investigating this with you we saw that it wasn't possible to get ``DateUtils`` to load without RequireJS. I was concerned about bundling, but it doesn't matter since ORA does its own bundling, and the three dependencies will all be pre-loaded in the platform.